### ISSUE TITLE
[18 regression] false positive `-Wmissing-field-initializer` warning for anonymous union with default member initializer

### ISSUE THREAD

**Issue by zygoloid** [llvm/llvm-project#70384]
This code:
```c++
struct A {
    int m;
    union { int n = 0; };
};

A a = A{.m = 0};
```
produces this false-positive warning:
```console
<source>:6:15: warning: missing field 'A::(anonymous union at <source>:3:5)' initializer [-Wmissing-field-initializers]
    6 | A a = A{.m = 0};
      |               ^
```

It looks like this regressed in https://reviews.llvm.org/D157879 / 91088978d712cd7b33610c59f69d87d5a39e3113.

**Comment by shafik** [comment]
CC @Fznamznon

**Comment by Fznamznon** [comment]
Oops, I missed there is already an issue and created my own https://github.com/llvm/llvm-project/issues/70444 . My bad.

**Comment by EugeneZelenko** [comment]
Not merged yet.

**Comment by chandlerc** [comment]
@zygoloid -- any suggestions on what we should actually do here?
